tests/contrib/langchain/test_langchain.py

```diff
@@ -1380,39 +1443,155 @@ def test_llmobs_huggingfacehub_llm(self, langchain, mock_llmobs_writer, mock_tra
+                        "output_value": str(
+                            {
+                                "question": "what is two raised to the fifty-fourth power?",
+                                "stop": ["```output"],
+                                "text": '\n```text\n2**54\n```\n...numexpr.evaluate("2**54")...\n',
+                            }
+                        ),
+                    },
+                ),
+                ("llm", {"provider": "openai", "input_role": None, "output_role": None}),
+            ],
+            different_py39_cassette=True,
+        )
+
+    @pytest.mark.skipif(sys.version_info < (3, 10, 0), reason="Requires unnecessary cassette file for Python 3.9")
+    def test_llmobs_chain_nested(self, langchain, mock_llmobs_writer, mock_tracer, request_vcr):
+        template = """Paraphrase this text:
+
+            {input_text}
+
+            Paraphrase: """
+        prompt = langchain.PromptTemplate(input_variables=["input_text"], template=template)
+        style_paraphrase_chain = langchain.chains.LLMChain(
+            llm=langchain.llms.OpenAI(model="gpt-3.5-turbo-instruct"), prompt=prompt, output_key="paraphrased_output"
+        )
+        rhyme_template = """Make this text rhyme:
+
+            {paraphrased_output}
+
+            Rhyme: """
+        rhyme_prompt = langchain.PromptTemplate(input_variables=["paraphrased_output"], template=rhyme_template)
+        rhyme_chain = langchain.chains.LLMChain(
+            llm=langchain.llms.OpenAI(model="gpt-3.5-turbo-instruct"), prompt=rhyme_prompt, output_key="final_output"
+        )
+        sequential_chain = langchain.chains.SequentialChain(
+            chains=[style_paraphrase_chain, rhyme_chain],
+            input_variables=["input_text"],
+            output_variables=["final_output"],
+        )
+
+        input_text = """
+            I have convinced myself that there is absolutely nothing in the world, no sky, no earth, no minds, no
+            bodies. Does it now follow that I too do not exist? No: if I convinced myself of something then I certainly
+            existed. But there is a deceiver of supreme power and cunning who is deliberately and constantly deceiving
+            me. In that case I too undoubtedly exist, if he is deceiving me; and let him deceive me as much as he can,
+            he will never bring it about that I am nothing so long as I think that I am something. So after considering
+            everything very thoroughly, I must finally conclude that this proposition, I am, I exist, is necessarily
+            true whenever it is put forward by me or conceived in my mind.
+            """
+
+        self._test_llmobs_chain_invoke(
+            generate_trace=lambda prompt: sequential_chain.run({"input_text": input_text}),
+            request_vcr=request_vcr,
+            mock_llmobs_writer=mock_llmobs_writer,
+            mock_tracer=mock_tracer,
+            cassette_name="openai_sequential_chain.yaml",
```

Comment:
Any chance we can just reuse the old `openai_sequential_paraphrase_and_rhyme_sync.yaml` cassette instead of creating a new `openai_sequential_chain.yaml`? Would be 200 less lines of code to maintain 😄

Reply:
I can try again, but for some reason it was giving cassette errors. will try again locally and in CI and lyk

Reply:
ah ok seems fine now, remove cassette 😄